### PR TITLE
lkl: add lkl_fdopendir and lkl_rewinddir

### PR DESCRIPTION
**Pull request by M1cha**
This change is [Reviewable](https://reviewable.io/reviews/lkl/linux/244)

### CODE REVIEW

File: tools/lkl/lib/fs.c [hunk 1]
```diff
@@ -276,6 +276,30 @@ struct lkl_dir *lkl_opendir(const char *path, int *err)
...
+		return NULL;
+	}
+
+	dir->fd = fd;
+
+	dir->len = 0;
+	dir->pos = NULL;
+
+	return dir;
+}
```
Comment (tavip) [inline tools/lkl/lib/fs.c]: There is some code we can share between lkl_fdopendir and lkl_opendir. Maybe add a new function lkl_dir_alloc that we can use in both?

**Comment by M1cha** [comment]
@tavip do you receive notifications now, or do you want me to write a comment after every change?

**Comment by M1cha** [comment]
pls don't merge it yet, I found a (stupid) bug

**Comment by M1cha** [comment]
ok fixed, lseek -> lkl_sys_lseek :P

**Comment by tavip** [comment]
Oops, I missed it during review :) Looks good, I will merge the two PRs tonight, thanks!

**Pull request merged**